File: Container_With_Most_Water.cpp
/*
 * Container_With_Most_Water.cpp
 * Copyright (C) 2014 fuel <fuel@fuel-3330>
 *
 * Distributed under terms of the MIT license.
 */

#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    int maxArea(vector<int> &height) {
        int L=0, R=height.size()-1, maxWater=0; 
        while(L<R){
            int barH=min(height[L],height[R]);
            int water = barH*(R-L);
            maxWater = max(maxWater, water);
            if(height[L]<height[R]) L++;
            else R--;
        }
        return maxWater;
    }
};

File: Container_With_Most_Water.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-
# vim:fenc=utf-8
#
# Copyright © 2014 fuel <fuel@fuel-3330>
#
# Distributed under terms of the MIT license.
class Solution:
    # @return an integer
    def maxArea(self, height):
        L,R,maxWater  = 0, len(height)-1, 0
        while(L<R):
            barH = min(height[L],height[R])
            water = (R-L)*barH
            maxWater = max(maxWater,water)
            if height[L]<height[R]:
                L+=1
            else:
                R-=1
        return maxWater
